### ISSUE TITLE
GameGuru MAX - Request - Windowed Mode for Test Game

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2787]
I'd like to request a Windowed mode for test game. This would be were I could run the test game in a window at a specified size, such as 1920x1080, etc. The main purpose of this window would be for people creating tutorials and the like to be able to capture video and get screen shots at specific sizes. I have an ultrawide monitor and MAX forces me to run at full-screen. So, any video captured is ultrawide, too.

**Comment by Necrym59** [comment]
I second this awsome idea for that and lot of other functional reasons, Hopping in and out to test a scipt is painful, it has to load the map then give errors then wait till you have oppportunity to close it (if it doesnt crash) then fix script error and reload/repeat, annoying at best. If this all happened in a window then if it crashes/errors it just goes back to editor, i can live with that.

**Comment by ArgentArts** [comment]
With other game engines, you can select to test your game IN THE VIEW PORT itself (I'd love this in MAX, actually), at FULL SCREEN, or IN A WINDOW. I definitely want to test in a window (as stated above), but it would be great to test in the editor itself, too. :)

**Comment by Necrym59** [comment]
Yeah in the viewport would be great too, although if it crashes doing scripts its no good but having those windowed/in-editor/fullscreen options would be great.

**Comment by ArgentArts** [comment]
Options are always good. You'd use windowed mode for both of the reasons we've outlined (for video capturing at a specific size and for code testing, anticipating possible crashes, etc.), full-screen mode to test the game to gauge how well it truly runs (since that's when it will it's best), and in the view port so you can do things like play the game to a certain spot, stop, make adjustments, play on, stop, fix this and that, add this element, play on, etc. ;)

**Comment by Necrym59** [comment]
I agree its important tool for game development for a ton of practical reasons, hopefully Lee and team will pick up on it, i suppose they arnt as focussued on these things at the moment as much as their focus is on developing a product and getting things working as opposed to a game developers point of view of useage.
